main/webservices/api/test_api.php

```diff
@@ -2,116 +2,564 @@
 /* For licensing terms, see /license.txt */
 
 /**
- * Test example to user API v2.php
+ * Test case for v2.php
  *
  * Using Guzzle' HTTP client to call the API endpoint and make requests.
- * Change URL on the first lines of createUser() below to suit your needs.
  */
 
 use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
 
-require_once '../../../vendor/autoload.php';
-
-// set your URL, username and password here to use it for all webservices in this test file.
-$webserviceURL='http://MYCHAMILO.loc/main/webservices/api/';
-$webserviceUsername='admin';
-$webservicePassword='admin';
+require_once __DIR__.'/../../../vendor/autoload.php';
+require_once __DIR__.'/../../inc/global.inc.php';
 
 /**
- * Make a request to get the API key for admin user.
- *
- * @throws Exception
- *
- * @return string
+ * Class V2Test
  */
-function authenticate() {
-    global $webserviceURL;
-    global $webserviceUsername;
-    global $webservicePassword;
-    $client = new Client([
-        'base_uri' => $webserviceURL,
-    ]);
-
-    $response = $client->post('v2.php', [
-        'form_params' => [
-            'action' => 'authenticate',
-            'username' => $webserviceUsername,
-            'password' => $webservicePassword,
-        ],
-    ]);
-
-    if ($response->getStatusCode() !== 200) {
-        throw new Exception('Entry denied with code : ' . $response->getStatusCode());
+class V2Test extends TestCase
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class V2Test has 11 non-getter- and setter-methods. Consider refactoring V2Test to keep number of methods under 10.](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)